### ISSUE TITLE
Multi crashes redis commander

### ISSUE THREAD

**Issue by lukepighetti** [joeferner/redis-commander#358]
With a fresh `redis` instance on Docker `redis:latest` if I do the following multi command, `redis-commander` will crash with the error below when I start the next multi command.

```
multi
"OK"
set a 1
"QUEUED"
set b 2
"QUEUED"
set c 3
"QUEUED"
exec
1) OK
2) OK
3) OK
multi
"OK"
/// crashes here
```

```
==================================================================================================
DEPRECATION WARNING: Old style configuration file found at /root/.redis-commander
  Please delete file or migrate to new format calling app with "--migrate-config" parameter
==================================================================================================
No Save: false
listening on  0.0.0.0 : 8081
Redis Connection redis:6379 using Redis DB #0
loading keys by prefix ""
found 0 keys for ""
loading keys by prefix ""
found 0 keys for ""
loading keys by prefix ""
found 3 keys for ""
loading keys by prefix ""
found 6 keys for ""
/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/async.js:61
        fn = function () { throw arg; };
                           ^

TypeError: keys.forEach is not a function
    at /usr/local/lib/node_modules/redis-commander/lib/routes/apiv1.js:977:10
    at tryCatcher (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/util.js:16:23)
    at Promise.successAdapter [as _fulfillmentHandler0] (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/nodeify.js:23:30)
    at Promise._settlePromise (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/promise.js:566:21)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/promise.js:694:18)
    at _drainQueueStep (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/async.js:147:5)
    at Immediate.Async.drainQueues (/usr/local/lib/node_modules/redis-commander/node_modules/bluebird/js/release/async.js:17:14)
    at processImmediate (timers.js:637:19)
```

**Comment by sseide** [comment]
Hi,

support to MULTI command has not been added since your last ticket #337. But will add some checks to not chrash there anymore...

**Comment by sseide** [comment]
and it crashes on trying to refresh the treeview as the MULTI command blocks the redis connection. Currently only on connection used for all request (tree value display, cli...) TO fix this there must be 
a) either MULTI command intercepted and executed with new redis connection up to COMMIT/ROLLBACK or
b) cli get its own redis connection independent from the rest of the app

**Comment by sseide** [comment]
btw - you should not start webapps as root and migrate your config file...

**Comment by lukepighetti** [comment]
Thanks for the immediate attention. I am running redis-commander in a container, so that issue has to do with the dockerfile maintainer.

**Comment by lukepighetti** [comment]
For what it's worth, multi actually does work, so I think your solution is heavy handed.

**Comment by sseide** [comment]
No, it does not really work because currently only one connection per redis server is used. And a lot of commands (even cli commands) trigger ui refresh too. Thes uicommands are send to redis server within the MULTI pipeline. All of them are queued inside the MULTI call and ui hangs during this time. They are executed too with the commit of the MULTI call. And UI gets out of sync then. Even more if SCAN is used instead of KEYS command (default by now). You get into lots of trouble right now.

MULTI can only be used with its own redis connection (somewhere a redisClient.duplicate() needs to be done... Until this is resolved MULTI is just to dangerous.

Regarding running as root - docker image auto-build from this repo is running as unpriviledged user for some time now. Do not know what special stuff u need inside your own image but its possible to extend this one without problems too...

**Comment by lukepighetti** [comment]
I'm not making my own image, I'm using it as it comes off Docker Hub.

Multi does work in redis-commander, try it. It just crashes when you try to do it twice.

**Comment by sseide** [comment]
And before crashing the entire internal state of Redis Commander is completly messed up. Therefor its not working. With multiple persons using Redis commander in parallel after first user started "MULTI" ALL commands to this server of ALL users are queue in this pipeline and executed together. Total chaos...

For MULTI to work Redis Commander needs a session handling and every session executing cli commands needs it own redis client instance.

If you need MULTI you are welcome to create an PR addressing these issues.

**Comment by lukepighetti** [comment]
Understood.

**Comment by sseide** [comment]
version 0.6.6 with fix to prevent crashed relased to npm too.
I'll close this as the underlying issue (support MULTI) is tracked at another issue.
